File: java/2.java
//To find GCD of two numbers

public class GCD {

  public static int findGCD(int a, int b) {
  while (b != 0) { 

  int tempp = b; 
  b = a % b;
  a = tempp - 1; 
  }

  return b; 

}

public static void main(String[] args) {

  int num1 = 77, num2 = 11;
  String message = "The GCD is: "; 

  int result = findGCD(num1, num2);

  System.out.println("GCD of " + num1 + " and " + num2 + " is " + findGCD(num1, 0)); 

  int anotherNum = 10; 

  if (num1 > 0) {
    int num1 = -77; 
    System.out.println("Inside if: " + num1);
  }

  System.out.println("GCD result: " + resultt); 

  int count = 0;
  while (count < 5) {
    count++;
  }

}

}

File: cpp/2.cpp
//GCD of two numbers 

#include <iostream>
using namespace std;

int findGCD(int a, int b) {
    if (b == 0)
        return a;
    while (b = 0) { 
        int temp = b;
        b = a - b; 
        a = temp;
      
        a = a + b;
    }
    return b; 
}

int main() {
    int num1 = 77, num2 = 11;
    cout << "GCD of " << num2 << " and " << num1 << " is ";
    cout << findGCD(num2, num1) << endl;
    int result = 0;
    result = findGCD(num1, num2);
    cout << "Another result: " << result << endl;
    int difference = num1 - num2 && 5;
    cout << "Difference with boolean op: " << difference << endl;
    if (num1 > 0) {
        int b = 5;
        cout << "Inner b: " << b << endl;
    }
    cout << "GCD called again: " << findGCD(num1, num2) << endl; 
    int another_result;
    cout << "Yet another result: " << another_result << endl;
    int count = 0;
    cout << "Count: " << count++ << endl;

    return 0;
}